src/interactive_cli/player_cli.py

```diff
@@ -0,0 +1,121 @@
+import functools
+from typing import Callable
+
+import inquirer
+
+from interactive_cli.cli import Cli
+
+
+class PlayerCli(Cli):
+
+    def __init__(self, message_call: Callable[[str], None]):
+        self.name = None
+        self.languages = []
+        self.message_call = message_call
+        self.current_language = None
+
+    def _is_char_configured(self):
+        if not self.name or not self.languages:
+            print("You want to play without a character? What are you, the director?")
+            return False
+        return True
+
+    def _confirm(self):
+        options = [
+            ('Yes', 'yes'),
+            ('No', 'no')
+        ]
+
+        answer = self.inquirer_prompt(
+            [inquirer.List('option', message="Are you sure of who you are?", choices=options, carousel=True)]
+        )
+
+        return answer['option'] == "yes"
+
+    def _configure_char(self):
+        print("Let's start by creating you character!\n")
+
+        questions = [
+            inquirer.Text(name='name', message='What\'s your name?'),
+            inquirer.Text(name='languages', message='What languages do you speak, {name}? (comma separated)'),
+        ]
+
+        answers = self.inquirer_prompt(questions)
+        self.name = answers['name']
+        self.languages = answers['languages'].split(',')
+
+        self._connect() if self._confirm() else self._configure_char()
+
+    def _connect(self):
+        self.message_call(f"{self.name}:{functools.reduce(lambda a, b: a + ',' + b, self.languages)}")
+        print("Welcome to the game!\n")
+
+    def _select_language(self):
+        if not self._is_char_configured():
+            return
+
+        answer = self.inquirer_prompt(
+            [inquirer.List('option', message="What language will you speak?", choices=self.languages, carousel=True)]
+        )
+
+        self.current_language = answer['option']
+
+    def _menu(self):
+        options = [
+            ('Select Language', 'language'),
+            ('Speak', 'speak')
+        ]
+
+        answer = self.inquirer_prompt(
+            [inquirer.List('option', message="Select Command", choices=options, carousel=True)]
+        )
+
+        match answer['option']:
+            case 'language':
+                self._select_language()
+                self._menu()
+            case 'speak':
+                return
+            case _:
+                print("I'm confused, you want to do what?\n")
+                self._menu()
+
+    def _speak(self, message):
+        if not self.current_language:
+            print("Can't speak without choosing which language you'll use now can you?\n")
+            return
+        self.message_call(f"{self.current_language}:{message}")
+
+    @staticmethod
+    def _print_help():
+        print()
+        print("\\menu       -> go to interactive menu")
+        print("\\language   -> select language")
```

Comment:
Maybe it would be cool termos estes comandos mais abstratos em vez de tao hard coded. Assim ate podiamos ter menus dentro de menus e seria mais facil adicionar um novo comando :eyes: . Que achas? Too much para este pr?

Reply:
não sei se percebi sequer xD

Reply:
Tipo termos uma classe para comandos para facilitar a criacao de novos comandos and all that

Reply:
ok, that sounds nice actually, mas secalhar fica para um refactor

Reply:
Agreed